test/src/test-cppapi-aggregates.cc

```diff
@@ -0,0 +1,448 @@
+  }
+}
+
+template <class T>
+std::string CppAggregatesFx<T>::write_sparse(
+    std::vector<T> a1,
+    std::vector<uint64_t> dim1,
+    std::vector<uint64_t> dim2,
+    uint64_t timestamp,
+    optional<std::vector<uint8_t>> a1_validity,
+    bool encrypt) {
+  // Open array.
+  std::unique_ptr<Array> array;
+  if (encrypt) {
+    array = std::make_unique<Array>(
+        ctx_,
+        SPARSE_ARRAY_NAME,
+        TILEDB_WRITE,
+        TemporalPolicy(TimeTravel, timestamp),
+        EncryptionAlgorithm(AESGCM, key_.c_str()));
+  } else {
+    array = std::make_unique<Array>(
+        ctx_,
+        SPARSE_ARRAY_NAME,
+        TILEDB_WRITE,
+        TemporalPolicy(TimestampStartEnd, 0, timestamp));
+  }
+
+  // Create query.
+  Query query(ctx_, *array, TILEDB_WRITE);
+  query.set_layout(TILEDB_GLOBAL_ORDER);
+  query.set_data_buffer("a1", a1);
+  query.set_data_buffer("d1", dim1);
+  query.set_data_buffer("d2", dim2);
+  if (a1_validity.has_value()) {
+    query.set_validity_buffer("a1", a1_validity.value());
+  }
+
+  // Submit/finalize the query.
+  query.submit();
+  query.finalize();
+
+  // Close array.
+  array->close();
+
+  return query.fragment_uri(0);
+}
+
+template <class T>
+void CppAggregatesFx<T>::remove_array(const std::string& array_name) {
+  if (!is_array(array_name))
+    return;
+
+  vfs_.remove_dir(array_name);
+}
+
+template <class T>
+void CppAggregatesFx<T>::remove_sparse_array() {
+  remove_array(SPARSE_ARRAY_NAME);
+}
+
+template <class T>
+bool CppAggregatesFx<T>::is_array(const std::string& array_name) {
+  return vfs_.is_dir(array_name);
+}
+
+TEST_CASE_METHOD(
+    CppAggregatesFx<int32_t>,
+    "C++ API: Aggregates basic count",
+    "[cppapi][aggregates][basic][count]") {
+  bool request_data = GENERATE(true, false);
+  bool allow_dups = GENERATE(true, false);
+  bool set_ranges = GENERATE(true, false);
+
+  create_sparse_array(allow_dups);
+
+  // Write fragments, only cell 3,3 is duplicated.
+  write_sparse({0, 1, 2, 3}, {1, 1, 1, 2}, {1, 2, 4, 3}, 1);
+  write_sparse({4, 5, 6, 7}, {2, 2, 3, 3}, {2, 4, 2, 3}, 3);
+  write_sparse({8, 9, 10, 11}, {2, 1, 3, 4}, {1, 3, 1, 1}, 4);
+  write_sparse({12, 13, 14, 15}, {4, 3, 3, 4}, {2, 3, 4, 4}, 6);
+
+  Array array{ctx_, SPARSE_ARRAY_NAME, TILEDB_READ};
+  Query query(ctx_, array, TILEDB_READ);
+
+  // Add a count aggregator to the query.
+  query.ptr()->query_->add_aggregator_to_default_channel(
+      "Count",
+      std::make_shared<tiledb::sm::CountAggregator>(
+          tiledb::sm::constants::all_attributes));
+
+  if (set_ranges) {
+    // Slice only rows 2, 3
+    Subarray subarray(ctx_, array);
+    subarray.add_range<uint64_t>(0, 3, 4);
+    query.set_subarray(subarray);
+  }
+
+  // Set the data buffer for the aggregator.
+  std::vector<uint64_t> count(1);
+  std::vector<uint64_t> dim1(100);
+  std::vector<uint64_t> dim2(100);
+  std::vector<int> a1(100);
+  query.set_layout(TILEDB_UNORDERED);
+  query.set_data_buffer("Count", count);
```

Comment:
Count will be an output field name that will need to be checked against existing names to avoid collision. The query object will make the differentiation.